application/src/main/java/run/halo/app/theme/endpoint/CommentFinderEndpoint.java

```diff
@@ -152,7 +165,37 @@ Mono<ServerResponse> createComment(ServerRequest request) {
                 comment.getSpec().setUserAgent(HaloUtils.userAgentFrom(request));
                 return commentService.create(comment);
             })
-            .flatMap(comment -> ServerResponse.ok().bodyValue(comment));
+            .transformDeferred(createIpBasedRateLimiter(request))
+            .flatMap(comment -> ServerResponse.ok().bodyValue(comment))
+            .onErrorResume(RequestNotPermitted.class, e -> ServerResponse.status(TOO_MANY_REQUESTS)
```

Comment:
IMO, the error handler here is redundant, because Halo has global error handler to handle RequestNotPermitted error, please refer to [here](https://github.com/halo-dev/halo/blob/e658097b9932b35bd3889ebe0fc0fa2f78f29ab2/application/src/main/java/run/halo/app/infra/exception/handlers/HaloErrorWebExceptionHandler.java).